backend/ticketvise/views/api/ticket.py

```diff
@@ -185,15 +185,21 @@ def get(self, request, inbox_id):
         labels = list(map(int, request.GET.getlist("labels[]", [])))
 
         inbox = get_object_or_404(Inbox, pk=inbox_id)
-        tickets = Ticket.objects.filter(inbox=inbox, title__icontains=q) | \
-                  Ticket.objects.filter(inbox=inbox, ticket_inbox_id__icontains=q).order_by("-date_created")
-
-        for term in q.split():
-            tickets = tickets | \
-                      (Ticket.objects.filter(author__username__icontains=term) |
-                       Ticket.objects.filter(author__first_name__icontains=term) |
-                       Ticket.objects.filter(author__last_name__icontains=term) |
-                       Ticket.objects.filter(author__email__contains=term))
+        if q:
+            # Only search when q exists to improve speed
+            users = User.objects.filter(inbox_relationship__inbox=inbox).search(q)
+
+            tickets = Ticket.objects.filter(inbox=inbox, title__icontains=q) | \
+                      Ticket.objects.filter(inbox=inbox, content__icontains=q) | \
+                      Ticket.objects.filter(inbox=inbox, ticket_inbox_id__icontains=q) | \
+                      Ticket.objects.filter(inbox=inbox, author__in=users) | \
+                      Ticket.objects.filter(inbox=inbox, assignee__in=users) | \
+                      Ticket.objects.filter(inbox=inbox, shared_with__in=users)
```

Comment:
Using [Q objects](https://docs.djangoproject.com/en/3.1/topics/db/queries/#complex-lookups-with-q-objects) would allow you to write this slightly less verbose, resulting in the same SQL:

```
tickets = Ticket.objects.filter(
    Q(title__icontains=q)
    | Q(content__icontains=q)
    | Q(ticket_inbox_id__icontains=q)
    | Q(author__in=users)
    | Q(assignee__in=users)
    | Q(shared_with__in=users),
    inbox=inbox,
)
```

Reply:
Thanks for the feedback. Using Q is indeed a neat solution. Combining this with SearchVector also seems to get a better result